cdap-api/src/main/java/co/cask/cdap/api/artifact/ArtifactVersion.java

```diff
@@ -30,11 +30,12 @@
  * parse version string in format of [major].[minor].[fix](-|.)[suffix]
  * }
  * </pre>
+ * Note that the suffix can not have numerical values in it.
  */
 @Beta
 public final class ArtifactVersion implements Comparable<ArtifactVersion> {
 
-  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?");
+  private static final Pattern PATTERN = Pattern.compile("(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-]([^\\d]*))?$");
```

Comment:
My understanding is that artifact name can also have numbers and dashes.
Because of this, if we allow suffix to have numbers, then isn't it ambiguous in this case:
"ali-1-1-foo-1"
Is the artifact name "ali", version is "1-1-foo-1"
or is the artifact name "ali-1" and the version is "1-foo-1"
or is the artifact name "ali-1-1-foo" and the version is "1"?

If we restrict suffix from having numbers, then its clear that the name is "ali-1-1-foo", and the version is "1"